fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/tenant/DeleteTenantKeyDAO.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.tenant;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import com.ibm.fhir.database.utils.api.IDatabaseStatement;
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+
+/**
+ * DAO to delete a tenant key record
+ */
+public class DeleteTenantKeyDAO implements IDatabaseStatement {
+    private final String schemaName;
+    private final int tenantId;
+    private final String tenantKey;
+    private int count = 0;
+
+    /**
+     * Public constructor
+     *
+     * @param schemaName
+     * @param tenantId
+     * @param tenantKey
+     */
+    public DeleteTenantKeyDAO(String schemaName, int tenantId, String tenantKey) {
+        DataDefinitionUtil.assertValidName(schemaName);
+        this.schemaName = schemaName;
+        this.tenantId = tenantId;
+        this.tenantKey = tenantKey;
+    }
+
+    @Override
+    public void run(IDatabaseTranslator translator, Connection c) {
+        final String tableName = DataDefinitionUtil.getQualifiedName(schemaName, "TENANT_KEYS");
+        final String SQL;
+        if (tenantKey == null) {
+            SQL = "DELETE FROM " + tableName + "  WHERE mt_id = ?";
+        } else {
+            SQL = "DELETE FROM " + tableName + "  WHERE mt_id = ?"
+                    + " AND tenant_hash = sysibm.hash(tenant_salt || ?, 2)";
```

Comment:
is this db2-specific.  if so, can it live in the translator?

Reply:
never mind, I think I was mixing up the translator with an "adapter" (where we have a different adapters for each db).  i think we prefer the adapter pattern where possible, but that this isn't compatible with our more generic IDatabaseStatement interface.

Reply:
Right, this is a db2 specific task - we're only multienant on db2.

Reply:
yeah, agree it should work just fine.  was just a general recommendation/observation to try keeping the db-specific logic in the adapter (in case we ever introduce any other multitenant dbs...which I don't think we will but :shrug: )

Reply:
maybe just add a comment to indicate that its db2-specific and move on?

Reply:
I started from AddTenantKeyDAO, this is consistent with addTenantKeyDAO and the other tenant daos. Are you asking me to update the comments in all of these?

Reply:
> I started from AddTenantKeyDAO, this is consistent with addTenantKeyDAO and the other tenant daos. Are you asking me to update the comments in all of these?

ah, I was wondering just that (how many places we've built in multitenant==db2 logic).  I think it would be good to call out all such places, but I guess its really not a big deal.

Reply:
I updated the top-level comments.